test/worker_test.rb

```diff
@@ -660,14 +660,35 @@ def self.perform
     assert_equal 1, Resque.workers.size
   end
 
-  it "does return returns a valid time when asking for heartbeat" do
+  it "does return a valid time when asking for heartbeat" do
     workerA = Resque::Worker.new(:jobs)
     workerA.register_worker
     workerA.heartbeat!
 
     assert_instance_of Time, workerA.heartbeat
   end
 
+  it "does not generate heartbeats that depend on the worker clock, but only on the server clock" do
+    server_time = Resque::Worker.server_time
+    fake_time = Time.parse("2000-01-01")
+
+    with_fake_time(fake_time) do
+      worker_time = Time.now
+      assert_equal fake_time, worker_time
```

Comment:
Isn't this assertion really a test for with_fake_time?  Should we test our test helpers like this in a separate test?